x-pack/platform/plugins/shared/ml/public/application/components/annotations/annotations_table/annotations_table.test.js

```diff
@@ -31,19 +39,59 @@ jest.mock('../../../services/ml_api_service', () => {
   };
 });
 
+jest.mock('@elastic/eui', () => {
+  const original = jest.requireActual('@elastic/eui');
+  return {
+    ...original,
+    EuiInMemoryTable: jest.fn().mockImplementation(({ items }) => {
+      return mockReact.createElement(
+        'div',
+        { 'data-test-subj': 'mockEuiInMemoryTable' },
+        `Mocked table with ${items?.length || 0} items`
+      );
+    }),
+  };
+});
+
+const mockKibanaContext = {
+  services: {
+    mlServices: {
+      mlApi: {},
+    },
+  },
+};
+
+// Mock withKibana HOC
+jest.mock('@kbn/kibana-react-plugin/public', () => {
```

Comment:
I found several `createWithKibanaMock` in the codebase. I reckon it'd be convenient to create a reusable helper in `kibana_react`